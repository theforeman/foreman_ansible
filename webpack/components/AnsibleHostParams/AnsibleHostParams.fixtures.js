export const rectangleLookupKey = {
  parameter: 'rectangle',
  id: 66,
  override: true,
  description: 'Rectangle param',
  parameterType: 'integer',
  omit: null,
  defaultValue: 17,
  hiddenValue: false,
  currentOverride: {
    value: null,
    element: 'hostgroup',
    elementName: 'first hostgroup',
  },
  overrideValues: [
    {
      id: 31,
      match: 'hostgroup=first hostgroup',
      omit: true,
      value: null,
    },
  ],
};

export const squareLookupKey = {
  parameter: 'square',
  id: 65,
  override: true,
  description: 'Square param',
  parameterType: 'boolean',
  omit: null,
  defaultValue: true,
  hiddenValue: false,
  currentOverride: {
    value: false,
    element: 'hostgroup',
    elementName: 'first hostgroup',
  },
  overrideValues: [
    {
      id: 26,
      match: 'hostgroup=first hostgroup',
      omit: false,
      value: false,
    },
  ],
};

export const squareLookupValue = {
  defaultValue: true,
  element: 'hostgroup',
  elementName: 'first hostgroup',
  id: 26,
  omit: false,
  overriden: true,
  value: false,
};

export const ellipseLookupKey = {
  currentOverride: null,
  defaultValue: false,
  description: '',
  hiddenValue: true,
  id: 79,
  omit: null,
  override: true,
  overrideValues: [],
  parameter: 'ellipse',
  parameterType: 'boolean',
  required: false,
  validatorRule: null,
  validatorType: '',
};

export const ellipseLookupValue = {
  defaultValue: false,
  element: 'fqdn',
  omit: false,
  overriden: false,
  value: false,
};

export const triangleLookupKey = {
  parameter: 'triangle',
  id: 99,
  override: true,
  description: 'Triangle param',
  parameterType: 'integer',
  omit: null,
  defaultValue: 19,
  hiddenValue: false,
  currentOverride: {
    value: 1,
    element: 'hostgroup',
    elementName: 'first hostgroup',
  },
  overrideValues: [
    {
      id: 30,
      match: 'os=CentOS 7',
      omit: false,
      value: 7,
    },
    {
      id: 35,
      match: 'hostgroup=second hostgroup',
      omit: false,
      value: 9,
    },
    {
      id: 31,
      match: 'hostgroup=first hostgroup',
      omit: false,
      value: 8,
    },
    {
      id: 36,
      match: 'hostgroup=third hostgroup',
      omit: false,
      value: 3,
    },
    {
      id: 30,
      match: 'domain=example.com',
      omit: false,
      value: 89,
    },
  ],
};

export const roles = [
  {
    id: 3,
    name: 'aardvark.cube',
    ansibleVariables: [rectangleLookupKey, squareLookupKey],
  },
];
