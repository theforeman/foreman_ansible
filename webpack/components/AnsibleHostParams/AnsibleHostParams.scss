#inherited_ansible_variables {
  .overridden .param_name {
    font-weight: bold;
  }

  .btn-override {
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
  }
}

.override-param {
  text-decoration: line-through;

  a,
  i {
    text-decoration: none;
  }

  .btn[data-tag] {
    display: none;
  }
}