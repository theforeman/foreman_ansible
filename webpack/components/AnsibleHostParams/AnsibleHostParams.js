import React, { useEffect } from 'react';
import { Spinner, Icon } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

import AnsibleVariablesTableRows from './components/AnsibleVariablesTableRows';
import GenericPopover from './components/GenericPopover';

import './AnsibleHostParams.scss';

const AnsibleHostParams = ({
  loading,
  assignedRoles,
  resourceErrors,
  highlightTabErrors,
  formObject,
}) => {
  const popoverText = __(
    'Foreman will not send this paramenter in classification output'
  );

  const popoverTitle = __('Omit parameter from classification');

  const omitHelpButton = (
    <a href="#">
      <Icon type="pf" name="info" />
    </a>
  );

  useEffect(() => {
    highlightTabErrors();
  });

  return (
    <div>
      <fieldset>
        <h2> {__('Ansible Variables')}</h2>
        <Spinner loading={loading}>
          <table className="table table-fixed" id="inherited_ansible_variables">
            <thead className="white-header">
              <tr>
                <th className="col-md-2">{__('Ansible Role')}</th>
                <th className="col-md-2">{__('Name')}</th>
                <th className="col-md-2">{__('Type')}</th>
                <th className="col-md-5">{__('Value')}</th>
                <th className="col-md-1 ca">
                  {__('Omit')}{' '}
                  <GenericPopover
                    button={omitHelpButton}
                    popoverText={popoverText}
                    popoverTitle={popoverTitle}
                    popoverId="ansible-omit-help"
                  />
                </th>
              </tr>
            </thead>
            <tbody>
              <AnsibleVariablesTableRows
                assignedRoles={assignedRoles}
                resourceErrors={resourceErrors}
                formObject={formObject}
              />
            </tbody>
          </table>
        </Spinner>
      </fieldset>
      <br />
    </div>
  );
};

AnsibleHostParams.propTypes = {
  loading: PropTypes.bool.isRequired,
  assignedRoles: PropTypes.array.isRequired,
  resourceErrors: PropTypes.array,
  highlightTabErrors: PropTypes.func.isRequired,
  formObject: PropTypes.object.isRequired,
};

AnsibleHostParams.defaultProps = {
  resourceErrors: [],
};

export default AnsibleHostParams;
