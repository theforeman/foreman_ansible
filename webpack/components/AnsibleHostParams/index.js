import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import { deepPropsToCamelCase } from 'foremanReact/common/helpers';

import AnsibleHostParams from './AnsibleHostParams';
import {
  selectAssignedVariables,
  selectVariablesLoading,
  selectFormObject,
} from '../AnsibleRolesSwitcher/AnsibleRolesSwitcherSelectors';

const WrappedAnsibleHostParams = props => {
  const loading = useSelector(state => selectVariablesLoading(state));
  const assignedRoles = useSelector(state => selectAssignedVariables(state));
  const formObject = useSelector(state => selectFormObject(state));

  return (
    <AnsibleHostParams
      loading={loading}
      assignedRoles={assignedRoles}
      resourceErrors={deepPropsToCamelCase(props.lookup_values).filter(
        val => Object.keys(val.errors).length > 0
      )}
      formObject={formObject}
      highlightTabErrors={window.tfm.tools.highlightTabErrors}
    />
  );
};

WrappedAnsibleHostParams.propTypes = {
  lookup_values: PropTypes.array,
};

WrappedAnsibleHostParams.defaultProps = {
  lookup_values: [],
};

export default WrappedAnsibleHostParams;
