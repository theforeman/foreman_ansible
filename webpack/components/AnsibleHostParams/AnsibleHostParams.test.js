import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import AnsibleHostParams from './AnsibleHostParams';

const props = {
  resourceErrors: [],
  highlightTabErrors: () => {},
  formObject: {
    resourceName: 'hostgroup',
    resourceId: null,
  },
};

const fixtures = {
  'should render when loading': {
    loading: true,
    assignedRoles: [],
    ...props,
  },
  'should render when loaded': {
    loading: false,
    assignedRoles: [],
    ...props,
  },
};

describe('AnsibleHostParams', () =>
  testComponentSnapshotsWithFixtures(AnsibleHostParams, fixtures));
