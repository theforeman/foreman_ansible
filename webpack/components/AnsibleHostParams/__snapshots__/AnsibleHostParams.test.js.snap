// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleHostParams should render when loaded 1`] = `
<div>
  <fieldset>
    <h2>
       
      Ansible Variables
    </h2>
    <Spinner
      className=""
      inline={false}
      inverse={false}
      loading={false}
      size="md"
    >
      <table
        className="table table-fixed"
        id="inherited_ansible_variables"
      >
        <thead
          className="white-header"
        >
          <tr>
            <th
              className="col-md-2"
            >
              Ansible Role
            </th>
            <th
              className="col-md-2"
            >
              Name
            </th>
            <th
              className="col-md-2"
            >
              Type
            </th>
            <th
              className="col-md-5"
            >
              Value
            </th>
            <th
              className="col-md-1 ca"
            >
              Omit
               
              <GenericPopover
                button={
                  <a
                    href="#"
                  >
                    <Icon
                      name="info"
                      type="pf"
                    />
                  </a>
                }
                placement="top"
                popoverId="ansible-omit-help"
                popoverOverlay={null}
                popoverText="Foreman will not send this paramenter in classification output"
                popoverTitle="Omit parameter from classification"
                trigger="click"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <AnsibleVariablesTableRows
            assignedRoles={Array []}
            formObject={
              Object {
                "resourceId": null,
                "resourceName": "hostgroup",
              }
            }
            resourceErrors={Array []}
          />
        </tbody>
      </table>
    </Spinner>
  </fieldset>
  <br />
</div>
`;

exports[`AnsibleHostParams should render when loading 1`] = `
<div>
  <fieldset>
    <h2>
       
      Ansible Variables
    </h2>
    <Spinner
      className=""
      inline={false}
      inverse={false}
      loading={true}
      size="md"
    >
      <table
        className="table table-fixed"
        id="inherited_ansible_variables"
      >
        <thead
          className="white-header"
        >
          <tr>
            <th
              className="col-md-2"
            >
              Ansible Role
            </th>
            <th
              className="col-md-2"
            >
              Name
            </th>
            <th
              className="col-md-2"
            >
              Type
            </th>
            <th
              className="col-md-5"
            >
              Value
            </th>
            <th
              className="col-md-1 ca"
            >
              Omit
               
              <GenericPopover
                button={
                  <a
                    href="#"
                  >
                    <Icon
                      name="info"
                      type="pf"
                    />
                  </a>
                }
                placement="top"
                popoverId="ansible-omit-help"
                popoverOverlay={null}
                popoverText="Foreman will not send this paramenter in classification output"
                popoverTitle="Omit parameter from classification"
                trigger="click"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <AnsibleVariablesTableRows
            assignedRoles={Array []}
            formObject={
              Object {
                "resourceId": null,
                "resourceName": "hostgroup",
              }
            }
            resourceErrors={Array []}
          />
        </tbody>
      </table>
    </Spinner>
  </fieldset>
  <br />
</div>
`;
